server/sonar-web/src/main/js/apps/webhooks/components/CreateWebhookForm.tsx

```diff
@@ -0,0 +1,193 @@
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import Modal from '../../../components/controls/Modal';
+import ModalValidationField from '../../../components/controls/ModalValidationField';
+import { Webhook } from '../../../app/types';
+import { translate } from '../../../helpers/l10n';
+
+interface Props {
+  onClose: () => void;
+  onDone: (data: { name: string; url: string }) => Promise<void>;
+  webhook?: Webhook;
+}
+
+interface State {
+  name: string;
+  nameError?: string;
+  submitting: boolean;
+  url: string;
+  urlError?: string;
+}
+
+export default class CreateWebhookForm extends React.PureComponent<Props, State> {
+  mounted: boolean;
+
+  constructor(props: Props) {
+    super(props);
+    const { webhook } = props;
+    const name = webhook ? webhook.name : '';
+    const url = webhook ? webhook.url : '';
+    this.state = {
+      name,
+      nameError: this.validateName(name),
+      submitting: false,
+      url,
+      urlError: this.validateUrl(url)
+    };
+  }
+
+  componentDidMount() {
+    this.mounted = true;
+  }
+
+  componentWillUnmount() {
+    this.mounted = false;
+  }
+
+  handleCancelClick = (event: React.SyntheticEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    event.currentTarget.blur();
+    this.props.onClose();
+  };
+
+  handleNameChange = (name: string) => this.setState({ name, nameError: this.validateName(name) });
+
+  handleUrlChange = (url: string) => this.setState({ url, urlError: this.validateUrl(url) });
+
+  handleFormSubmit = (event: React.SyntheticEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (this.isValid()) {
+      this.setState({ submitting: true });
+      this.props
+        .onDone({ name: this.state.name, url: this.state.url })
+        .then(this.props.onClose, () => {
+          if (this.mounted) {
+            this.setState({ submitting: false });
+          }
+        });
+    }
+  };
+
+  isValid = () => !this.state.nameError && !this.state.urlError;
+
+  validateName = (name: string) => {
+    if (!name.trim()) {
+      return translate('webhooks.name.required');
+    }
+    return undefined;
+  };
+
+  validateUrl = (url: string) => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this function to use "return" consistently ([typescript:S3801](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3801))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-647%2FSONAR-10345_manage_webhooks&issues=AWFmrcL8ZGaLdh2cMSV5&open=AWFmrcL8ZGaLdh2cMSV5)

[ISSUE_KEY]: AWFmrcL8ZGaLdh2cMSV5